### DISCUSSION TITLE
404 while font loading

### DISCUSSION THREAD

**Discussion by flappix** [spatie/laravel-google-fonts#26] [Q&A]
Hi,
 this is my ```config/google-fonts.php```

```
<?php

return [
    'fonts' => [
        'default' => 'https://fonts.googleapis.com/css2?family=Manrope&display=swap',
    ],
    'disk' => 'public',
    'path' => 'fonts',
    'inline' => true,
    'fallback' => ! env('APP_DEBUG'),
    'user_agent' => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15',

];
```

In my ```.env``` I set
```
APP_URL=http://localhost/myproject
```

My test blade file looks like this:
```
<!DOCTYPE html>
<html lang="en">
	<head>
		@googlefonts ('default')
		<style type="text/css">
			h1
			{
				font-family: Manrope;
			}
		</style>
	</head>
	<body>
		<h1>Test</h1>
	</body>
</html>
```

I run ```php artisan google-fonts:fetch``` and confirmed that the files has been downloaded
```
> ls -R storage/app/public/fonts 
storage/app/public/fonts:
988bc0f675

storage/app/public/fonts/988bc0f675:
fonts.css
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-b-b-tfc7agrajwa.woff
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-c-b-tfc7agry.woff
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-g-b-tfc7agrajwa.woff
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-m-b-tfc7agrajwa.woff
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-n-b-tfc7agrajwa.woff
smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-p-b-tfc7agrajwa.woff
```

When I call the url to my blade file the browser tries to loads the font from this url, which fails (404)
```http://localhost/myproject/storage/fonts/988bc0f675/smanropev13xn7-yhe41ni1adirqauzuw1bx9mbzk79fn-c-b-tfc7agry.woff```

Any idea what I'm doing wrong?

**Laravel Version**: 8.54
**laravel-google-fonts Version**: 1.2

**Comment by flappix** [comment]
I found a solution some minutes after posting this question.   
I created a symlink from storage/app/public/ to public/storage. Now it works. Is this the intended way of using this library? If so maybe a hint in the installation instructions might be helpful.

**Reply by rapgithub** [reply]
php artisan storage:link  create a connection from public/storage to storage/app/public/  so how did you created that symlink? in my case still fonts not found!
any ideas?

**Comment by Kristories** [comment]
Basically, if you choose `public` disk, then without further ado, you should know how it works (with `storage:link`), as it's explained in the Laravel documentation.

In the README ([config section](https://github.com/spatie/laravel-google-fonts/blob/main/README.md?plain=1#L85)), there is also an explanation in the comments

```
/*
 * This disk will be used to store local Google Fonts. The public disk
 * is the default because it can be served over HTTP with storage:link.
 */
    'disk' => 'public',
```

**Comment by rapgithub** [comment]
I FOLLOWED ALL STEPS AND STILL FONTS NOT FOUND! ANY IDEAS? THANKS
AND FONTS ARE already in storage/app/fonts anyway!

any idea? thanks
